src/secondary_threads/parse_packets.rs

```diff
@@ -45,152 +50,206 @@ pub fn parse_packets(
                                             exchanged_bytes,
                                             new_info.traffic_direction,
                                         );
                                     });
-                                }
                             }
-
-                            //increment the packet count for the sniffed app protocol
-                            info_traffic
-                                .app_protocols
-                                .entry(application_protocol)
-                                .and_modify(|data_info| {
-                                    data_info
+                            (_, true) => {
+                                // rDNS already resolved
+                                // update the corresponding host's data info
+                                let host = info_traffic
+                                    .addresses_resolved
+                                    .get(&address_to_lookup)
+                                    .unwrap()
+                                    .1
+                                    .clone();
+                                info_traffic.hosts.entry(host).and_modify(|data_info_host| {
+                                    data_info_host
+                                        .data_info
                                         .add_packet(exchanged_bytes, new_info.traffic_direction);
-                                })
-                                .or_insert(DataInfo::new_with_first_packet(
-                                    exchanged_bytes,
-                                    new_info.traffic_direction,
-                                ));
+                                });
+                            }
                         }
+
+                        //increment the packet count for the sniffed app protocol
+                        info_traffic
+                            .app_protocols
+                            .entry(application_protocol)
+                            .and_modify(|data_info| {
+                                data_info.add_packet(exchanged_bytes, new_info.traffic_direction);
+                            })
+                            .or_insert(DataInfo::new_with_first_packet(
+                                exchanged_bytes,
+                                new_info.traffic_direction,
+                            ));
                     }
                 }
             }
         }
     }
 }
+
+fn get_sniffable_headers<'a>(
+    packet: &'a Packet,
+    link_type: &mut Linktype,
+    info_traffic_mutex: &Arc<Mutex<InfoTraffic>>,
+    is_link_type_confirmed: &mut bool,
+) -> Result<PacketHeaders<'a>, ReadError> {
+    match is_link_type_confirmed {
+        false => {
+            let ethernet_result = PacketHeaders::from_ethernet_slice(packet);
+            let ip_result = PacketHeaders::from_ip_slice(packet);
+            let null_result = PacketHeaders::from_ip_slice(&packet[4..]);
```

Comment:
(same comment as bellow)

```suggestion
            let null_result = try_from_null_frame(packet);
```